modules/programs/emacs.nix

```diff
@@ -3,6 +3,7 @@
 with lib;
 
 let
+  types = lib.types // import ../lib/types.nix { inherit lib; };
```

Comment:
Better to add a `types` attribute in `modules/lib/default.nix` that imports `types.nix`. Then remove this line and use, e.g., `type = config.lib.types.selectorFunction;` instead. I _think_ this should work.

If that doesn't work then I would accept doing the import here but it seem unnecessary to merge with `lib.types`. Could have
```
hmTypes = import ../lib/types.nix { inherit lib; }
```
or something similar.

Reply:
Had another look at the code and I suspect it wouldn't work to have ` = config.lib.types.selectorFunction;` 😕

Reply:
Yah, not entirely sure how to handle this properly. I don't like what I did here either..
Though my hope was that nixpkgs would actually get this type merged, there is a PR https://github.com/NixOS/nixpkgs/pull/44601